### PR TITLE
Email test cases

### PR DESCRIPTION
**Pull request by YakMan101**
Added test cases to the daily-emailing and weekly-emailing folders.

Also refactored/created new functions to make the testing easier 
and the adjusted the scripts themselves more consistent.

### CODE REVIEW

File: terraform/variables.tf [hunk 1]
```diff
@@ -29,5 +29,4 @@ variable "DB_NAME" {type = string}
 
 variable "FROM_EMAIL" {type = string}
 
-variable "OPENAI_API_KEY" {type = string}
```
Comment (ayahusseini) [inline terraform/variables.tf:32]: good catch removing unneeded variables

File: news_sentiment_analyser/README.md [hunk 2]
```diff
@@ -35,7 +35,6 @@ Optional:
 
     # OpenAI Configuration
     OPENAI_API_KEY=<your_openai_key>
-    OPENAI_MODEL=<your-openai-model>
```
Comment (ayahusseini) [inline news_sentiment_analyser/README.md:38]: good correction to README.md

**Review by ayahusseini — APPROVED** [review]
Looks good, thanks for removing the openai model

File: daily-emailing/d_db_funcs.py [hunk 3]
```diff
@@ -20,9 +23,22 @@ def create_connection() -> connection:
     return conn
 
 
+def get_cursor(conn: connection) -> cursor:
+    """Return cursor object to execute sql commands"""
+
+    return conn.cursor()
+
+
+def get_yesterday_date() -> str:
```
Comment (indiah444) [inline daily-emailing/d_db_funcs.py:32]: Good idea to refactor this into a function.

File: daily-emailing/daily_email.py [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
 
 from dotenv import load_dotenv
 import boto3
 
 from html_content import generate_html
-from d_db_funcs import get_avg_polarity_by_topic_and_source_yesterday, get_daily_subscribers
+from d_db_funcs import get_avg_polarity_by_topic_and_source_yesterday, get_daily_subscribers, get_yesterday_date
+
+
+def get_ses_client():
```
Comment (indiah444) [inline daily-emailing/daily_email.py:14]: Another good reusable function!

File: daily-emailing/test_d_db_funcs.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+from unittest.mock import MagicMock, patch
+from datetime import datetime, timedelta
+
+import pandas as pd
+
+from d_db_funcs import get_avg_polarity_by_topic_and_source_yesterday, get_daily_subscribers, get_yesterday_links
+
+
+class TestGetAvgPolarityByTopicAndSourceYesterday:
```
Comment (indiah444) [inline daily-emailing/test_d_db_funcs.py:11]: Love test classes :)

File: daily-emailing/test_daily_email.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+from daily_email import send_email, lambda_handler
+
+
+@patch('daily_email.get_daily_subscribers')
+@patch('daily_email.get_avg_polarity_by_topic_and_source_yesterday')
+@patch('daily_email.generate_html')
+@patch('daily_email.get_ses_client')
+@patch('daily_email.MIMEMultipart')
+@patch.dict('os.environ', {'FROM_EMAIL': 'from@example.com'})
```
Comment (indiah444) [inline daily-emailing/test_daily_email.py:13]: Good patching in environment variables

File: weekly-emailing/conftest.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# pylint: skip-file
+
+import pytest
+import pandas as pd
+import altair as alt
+
+
+@pytest.fixture
```
Comment (indiah444) [inline weekly-emailing/conftest.py:8]: Nice use of fixtures!

**Pull request merged**